### PR TITLE
Using the Analog Discovery Guide

### PR DESCRIPTION
**Pull request by psychogenic**
Uuuf: a guide for the AD3.

### CODE REVIEW

File: content/guides/analog-discovery/_index.md [hunk 1]
```diff
@@ -0,0 +1,886 @@
...
+
+
+There is a sample *WaveForms* workspace that is ready to use that can 
+   * talk to the demoboard REPL over serial to setup the correct project;
+   * drive the demoboard inputs to dynamically control the project;
+   * monitor and measure the ring oscillator and DAC outputs; and
+   * graph the results and output CSV files with data.
+
+
+The purpose here is to walk-through the entire process while introducing various features of the AD3 and some of the ways to use it.  If you just want the bullet point list of steps without any details, skip to the end of this page.
```
Comment (mattvenn) [inline content/guides/analog-discovery/_index.md:19]: the bullet point list is not at the end of the page, maybe you re-organised things?

File: content/guides/analog-discovery/_index.md [hunk 2]
```diff
@@ -0,0 +1,886 @@
...
+
+
+### WaveForms Installation
+
+To play along, an Analog Discovery 3 is required, as is its front-end application, "Waveforms".  Follow the [AD3 Getting Started Guide](https://digilent.com/reference/test-and-measurement/analog-discovery-3/getting-started#download_and_install_waveforms) to get everything setup and test that it is working.
+
+
+### Running The Full Test Suite
+
+We'll walk through every important step, below, and that is a good way to get to know the Analog Discovery.  However, if you want to just get going and explore things on your own, you may simply wire things up and run the pre-built workspace on the AD3, you may just follow the directions in this section.
```
Comment (mattvenn) [inline content/guides/analog-discovery/_index.md:29]: end of sentence is a bit clunky. how about:

"you can wire things up, load the provided workspace on the AD3 and follow the bullet point list of steps"

also that uses the same language as above so the user is more sure of the 2 options. bullet point or full walkthrough

File: content/guides/analog-discovery/_index.md [hunk 3]
```diff
@@ -0,0 +1,886 @@
...
+
+## What you need
+
+   1. An Analog Discovery 3 and its *WaveForms* program installed, connected to the computer;
+   2. The *tt05analogdemoFull.dwf3work* workspace opened in *WaveForms*;
+   3. A TT05 demoboard and ASIC, also connected to the computer via USB;
+   4. To have things wired up
```
Comment (mattvenn) [inline content/guides/analog-discovery/_index.md:36]: 4. The AD3 and demo board correctly wired

File: content/guides/analog-discovery/_index.md [hunk 4]
```diff
@@ -0,0 +1,886 @@
...
+```
+
+If you also want to run the oscillator frequency vs core voltage characterization, you'll need to
+
+   * depopulate F2 from the demoboard
+   * wire *V+* from the AD3 to the 1V8 pin on the demoboard top header
+
+
+This is optional as you may skip the test at the prompt
+{{< figure src="images/47.png" title="Test run prompt" >}}
```
Comment (mattvenn) [inline content/guides/analog-discovery/_index.md:62]: would remove the image and change to :This is optional and you can skip the test when prompted.

File: content/guides/analog-discovery/_index.md [hunk 5]
```diff
@@ -0,0 +1,886 @@
...
+
+or simply get the boring result of no-frequency-variation (since the core voltage won't actually be set by the AD3).
+
+## How to run
+
+
+Assuming you've wired everything up, these are the steps to run the test suite.  Following the entire sequence is only important if you've set things to drive the core supply from the AD3, but it doesn't hurt even if you haven't
```
Comment (mattvenn) [inline content/guides/analog-discovery/_index.md:69]: Remove 'assuming you've wired everything up' - as it's already part of the checklist

Comment (mattvenn) [reply content/guides/analog-discovery/_index.md:69]: add a : or a full stop at end of sentence

File: content/guides/analog-discovery/_index.md [hunk 6]
```diff
@@ -0,0 +1,886 @@
...
+
+   1. connect the AD3 to USB
+   2. open the WaveForms program and load the workspace file
+   3. switch to the *Supplies* tab, ensure Positive supply is set to 1.8V and press the "play" button on that tab to enable the core voltage.  Make sure both the "master enable" and "positive supply" have green checkmarks;
```
Comment (mattvenn) [inline content/guides/analog-discovery/_index.md:73]: the lists are mixed first capatilisation, and ; or nothing for ending.
pls stick to one style for the whole doc
Preferred: start with a capital, end with a ,

File: content/guides/analog-discovery/_index.md [hunk 7]
```diff
@@ -0,0 +1,886 @@
...
+   7. press "play" on the *powerupHighZ*, again... but this time click "No" to the first prompt and then "Yes" to "Do you wish to restore Patterns signals to PP?"
+
+Now, everything should be in a good state, and you shouldn't have to go through that dance again unless you power down.
+
+Clicking the "play" button on the *tests* tab in Script will run the tests--simply follow the prompts.
+
+{{< figure src="images/49.png" title="DAC plot prompt" >}}
+
+
+
```
Comment (mattvenn) [inline content/guides/analog-discovery/_index.md:86]: add a heading here like 'full walkthrough'. then that can also be linked to above when mentioning it.

File: content/guides/analog-discovery/_index.md [hunk 8]
```diff
@@ -0,0 +1,886 @@
...
+ * 2+ is connected to the *m35* pin (the DAC output); and 
+ * both 1- and 2- are connected to GND pads.
+ 
+
+
+{{< figure src="images/ad3scopeconn.jpg" title="AD3 Scope connections" >}}
+
+## Ring Oscillator 
+
+For this to work, the analog test project must loaded and configured.  This could be done through the [commander app](https://commander.tinytapeout.com/), but it's sometimes quicker to connect to the REPL directly via a serial terminal and use the SDK to:
```
Comment (mattvenn) [inline content/guides/analog-discovery/_index.md:111]: worth mentioning there are 2approaches taken here, the script talks directly to the board and you don't use the commander. the walkthrough uses more manual steps with the commander to configure things

File: content/guides/analog-discovery/_index.md [hunk 9]
```diff
@@ -0,0 +1,886 @@
...
+```
+ 
+{{< figure src="images/7.png" title="REPL interaction" >}}
+
+The ring oscillator should now be running, output on the pin connected to channel 1 and visible in the scope (click *Run*--the "play" icon--if you can't see it, or check the connections).
+ 
+{{< figure src="images/8.png" title="OSC on scope" >}}
+
+The scroll wheel to zoom in and drag the trigger level (yellow arrow on right side) to get something stable.
+Use the range on the right to get a better look by stretching it vertically, and drag the offest arrow on the left to position it.
```
Comment (mattvenn) [inline content/guides/analog-discovery/_index.md:131]: typo offest

File: content/guides/analog-discovery/_index.md [hunk 10]
```diff
@@ -0,0 +1,886 @@
...
+To look at the DAC function, channel 2 can be re-enabled using the checkbox and scaled an positioned just like C1.
+
+{{< figure src="images/18.png" title="Measurements made" >}}
+
+
+There's a good deal of cross-talk, likely owing to the dangly wires used to probe the analog pins, so the ring oscillator is disabled while the DAC is enabled, by setting the in3 bit and clearing the rest, in the REPL:
+
+```
+>>> tt.input_byte = 0
+>>> tt.in3(1)
```
Comment (mattvenn) [inline content/guides/analog-discovery/_index.md:178]: stick to one style of setting the pins? sometimes you use input_byte, sometimes in3 - a bit confusing

**Review by mattvenn — COMMENTED** [review]
wow long one! I have submitted a few comments on readability / style.
For me, the tests don't work, but the manual walkthrough method does. So let's troubleshoot the tests together sometime.

File: content/guides/analog-discovery/_index.md (hunk 1 above)
Comment (psychogenic) [reply content/guides/analog-discovery/_index.md:19]: I did.  Ok, that bug report for the workspace is lacking but I'll go over each of these and amend the PR.

**Comment by psychogenic** [comment]
Ok, made the mods, won't respond to each of these individually.  Check it when possible :)

**Comment by mattvenn** [comment]
Thanks!

**Comment by psychogenic** [comment]
Cool :) I'm going to go through this one more time from scratch, just to make sure I didn't flub the pin mapping or something, to see if I can replicate or resolve the issue you were having.

**Pull request merged**